x/staking/src/keeper/mod.rs

```diff
@@ -236,48 +226,53 @@ impl<
         }
         // if balance is different from non bonded coins panic because genesis is most likely malformed
         if not_bonded_balance != not_bonded_coins {
-            return Err(AppError::Custom(format!(
+            panic!(
```

Comment:
Panic?

Reply:
Here should be a panic because it's unfailable method. But yes, it's better to use `assert_eq`.
I'll check the other methods.